ex04.ex

```diff
@@ -67,15 +71,25 @@ defmodule Ex04 do
   `Integer.is_odd`, and you can use the `reverse` function you defined
   above. Feel free to write helper functions if you want.
 
-      iex> Ex04.even_odd [ 1, 2, 3, 4, 5 ]
+      iex> Ex04.even_odd([ 1, 2, 3, 4, 5 ])
       { [ 2, 4],  [ 1, 3, 5 ] }
 
   Hint: you're taking a list and converting it into something else. What function
   helps you do that. And, if you use that function, what does it return? That
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def purge_nil(list) do
```

Comment:
# 5

Although this works, it traverses the list twice, and it isn't very functional. I was looking for something like:

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &assign_one_even_odd/2)
    |> reverse_two_lists
  end

  defp assign_one_even_odd(number, { evens, odds }) do
    cond do
      Integer.is_even(number)->
        { [ number | evens ], odds }
      Integer.is_odd(number) ->
        { evens, [ number | odds ] }
    end
  end

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~